Source/missiles.cpp

```diff
@@ -2591,79 +2630,84 @@ void AddFirewallC(int mi, Point src, Point dst, int midir, int8_t /*mienemy*/, i
+	if (!missile._miDelFlag) {
+		missile._miVar7 = 0;
+		missile.limitReached = false;
+		missile._miVar3 = left[left[midir]];
+		missile._miVar4 = right[right[midir]];
+		missile._mirange = 7;
 		UseMana(id, SPL_FIREWALL);
 	}
 }
 
 void AddInfra(int mi, Point /*src*/, Point /*dst*/, int /*midir*/, int8_t mienemy, int id, int /*dam*/)
 {
-	Missiles[mi]._mirange = ScaleSpellEffect(1584, Missiles[mi]._mispllvl);
-	Missiles[mi]._mirange += Missiles[mi]._mirange * Players[id]._pISplDur / 128;
+	auto &missile = Missiles[mi];
+	missile._mirange = ScaleSpellEffect(1584, missile._mispllvl);
+	missile._mirange += missile._mirange * Players[id]._pISplDur / 128;
 
 	if (mienemy == TARGET_MONSTERS)
 		UseMana(id, SPL_INFRA);
 }
 
 void AddWave(int mi, Point /*src*/, Point dst, int /*midir*/, int8_t /*mienemy*/, int id, int /*dam*/)
 {
-	Missiles[mi]._miVar1 = dst.x;
-	Missiles[mi]._miVar2 = dst.y;
-	Missiles[mi]._miVar3 = 0;
-	Missiles[mi]._miVar4 = 0;
-	Missiles[mi]._mirange = 1;
-	Missiles[mi]._miAnimFrame = 4;
+	auto &missile = Missiles[mi];
+	missile._miVar1 = dst.x;
+	missile._miVar2 = dst.y;
+	missile._miVar3 = 0;
+	missile._miVar4 = 0;
+	missile._mirange = 1;
+	missile._miAnimFrame = 4;
 	UseMana(id, SPL_WAVE);
 }
 
 void AddNova(int mi, Point /*src*/, Point dst, int /*midir*/, int8_t mienemy, int id, int /*dam*/)
 {
-	Missiles[mi]._miVar1 = dst.x;
-	Missiles[mi]._miVar2 = dst.y;
+	auto &missile = Missiles[mi];
+	missile._miVar1 = dst.x;
+	missile._miVar2 = dst.y;
 
 	if (id != -1) {
 		int dmg = GenerateRndSum(6, 5) + Players[id]._pLevel + 5;
-		Missiles[mi]._midam = ScaleSpellEffect(dmg / 2, Missiles[mi]._mispllvl);
+		missile._midam = ScaleSpellEffect(dmg / 2, missile._mispllvl);
 
 		if (mienemy == TARGET_MONSTERS)
 			UseMana(id, SPL_NOVA);
 	} else {
-		Missiles[mi]._midam = (currlevel / 2) + GenerateRndSum(3, 3);
+		missile._midam = (currlevel / 2) + GenerateRndSum(3, 3);
 	}
 
-	Missiles[mi]._mirange = 1;
+	missile._mirange = 1;
 }
 
 void AddBlodboil(int mi, Point /*src*/, Point /*dst*/, int /*midir*/, int8_t /*mienemy*/, int id, int /*dam*/)
 {
 	auto &player = Players[id];
+	auto &missile = Missiles[mi];
 
 	if ((player._pSpellFlags & 6) != 0 || player._pHitPoints <= player._pLevel << 6) {
-		Missiles[mi]._miDelFlag = true;
+		missile._miDelFlag = true;
 		return;
 	}
 
 	UseMana(id, SPL_BLODBOIL);
-	Missiles[mi]._miVar1 = id;
+	missile._miVar1 = id;
 	int tmp = 3 * player._pLevel;
 	tmp <<= 7;
 	player._pSpellFlags |= 2;
-	Missiles[mi]._miVar2 = tmp;
+	missile._miVar2 = tmp;
 	int lvl = player._pLevel * 2;
-	Missiles[mi]._mirange = lvl + 10 * Missiles[mi]._mispllvl + 245;
+	missile._mirange = lvl + 10 * missile._mispllvl + 245;
 	CalcPlrItemVals(id, true);
 	force_redraw = 255;
 	player.Say(HeroSpeech::Aaaaargh);
 }
 
 void AddRepair(int mi, Point /*src*/, Point /*dst*/, int /*midir*/, int8_t /*mienemy*/, int id, int /*dam*/)
 {
-	Missiles[mi]._miDelFlag = true;
+	auto &missile = Missiles[mi];
+	missile._miDelFlag = true;

```

Comment:
I normally wouldn't do this chance if there was only one usage, but in the case of `void Add*(int mi,` we want to change them to take `MissileStruct` as the input, so it can server as prep for that still.